server/tests/external/c3/test_client.py

```diff
@@ -0,0 +1,384 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Written by:
+#        Nadzeya Hutsko <nadzeya.hutsko@canonical.com>
+
+from fastapi.testclient import TestClient
+from requests_mock import Mocker
+from sqlalchemy.orm import Session
+
+from hwapi.data_models import models
+from hwapi.external.c3.client import C3Client
+from tests.data_generator import DataGenerator
+
+
+def test_successful_load_certficates(
+    db_session: Session, requests_mock: Mocker, test_client: TestClient
+):
+    """Test that certificates and hardware data are imported correctly"""
+    requests_mock.get(
+        "https://c3_url/api/v2/public-certificates/",
+        json={
+            "count": 1,
+            "next": None,
+            "previous": None,
+            "results": [
+                {
+                    "canonical_id": "201806-26288",
+                    "vendor": "Dell",
+                    "platform": "Z8 G4 Workstation",
+                    "configuration": "Z8 G4 Workstation (i3)",
+                    "created_at": "2022-08-15T17:31:54.008447+01:00",
+                    "completed": "2022-08-15T17:31:54.124274+01:00",
+                    "name": "2208-11729",
+                    "release": {
+                        "codename": "bionic",
+                        "release": "18.04 LTS",
+                        "release_date": "2018-04-26",
+                        "supported_until": "2028-04-25",
+                        "i_version": 1804,
+                    },
+                    "architecture": "amd64",
+                    "kernel_version": "4.15.0-55-generic",
+                    "bios": {
+                        "name": "Dell Inc.: U32",
+                        "vendor": "Dell Inc.",
+                        "version": "U3",
+                        "firmware_type": "",
+                    },
+                    "firmware_revision": "1.40",
+                }
+            ],
+        },
+    )
+
+    c3client = C3Client(db=db_session)
+    c3client.load_certified_configurations()
+
+    # Verify vendors
+    assert db_session.query(models.Vendor).count() == 1
+    vendor = db_session.query(models.Vendor).filter_by(name="Dell").first()
+    assert vendor is not None
+
+    # Verify plarforms
+    platform = (
+        db_session.query(models.Platform).filter_by(name="Z8 G4 Workstation").first()
+    )
+    assert platform is not None
+    assert platform.vendor == vendor
```

Comment:
this is an old test, I've just renamed the file. I guess if we split these checks, we'll have a lot of duplicates in the code since we work with the same response

Reply:
> if we split these checks, we'll have a lot of duplicates in the code

sounds to me like we already have duplicates, it's just that splitting the code makes them more obvious. In that case wouldn't we be able to reuse functions and reduce duplication?

Reply:
I've simplified the assertions